### PR TITLE
Fix ocaml_light incompatibility with Coq 8.8, avoid outputting deprecated Coq code

### PR DESCRIPTION
**Pull request by palmskog**
The Ott Coq library works with the newly released [Coq 8.8](https://coq.inria.fr/news/145.html), but the `ocaml_light` example does not. This pull request fixes that, and also removes all (generation of) deprecated Coq code - in particular, using `Implicit Arguments` rather than just `Arguments`.

There is also one small quality-of-developer-life change: I added the installation directives from the OPAM package as the file `ott.install`, which makes it easy to install experimental versions of Ott locally by pinning (this has been indispensable for me in testing). If this is controversial, I can remove remove this change.

Coq library and `ocaml_light` Coq code tested with: Coq 8.5.3, Coq 8.6.1, Coq 8.7.2, and Coq 8.8.0.

### CODE REVIEW

**Comment by PeterSewell** [comment]
thanks!

File: ott.install [hunk 1]
```diff
@@ -0,0 +1,3 @@
+bin: ["src/ott"]
+doc: ["doc/ott_manual_0.25.pdf" "doc/ott_manual_0.25.html"]
```
Comment (hannesm) [inline ott.install]: the `doc/ott_manual_0.25.*` are no longer around, I adjusted it in eb88000 to use `built_doc/top2.*`

**Comment by palmskog** [comment]
@hannesm oh right, thanks for catching that.

**Comment by hannesm** [comment]
@palmskog what was the motivation to move `opam` to `ott.opam`?

**Comment by palmskog** [comment]
Isn't the convention if you have multiple packages with different metadata to have them be called `pkg1.opam`, `pkg1.install`, `pkg2.opam`, etc.? I also wasn't sure what the `.install` file should be called if there was only an `opam` file (and not `ott.opam`). Feel free to change this back if you want.

**Comment by palmskog** [comment]
For example, this convention is used here: https://github.com/dbuenzli/topkg

**Comment by hannesm** [comment]
@palmskog ah, yes, that's fine.  thanks!

**Pull request merged**